src/executables/master_heartbeat.c

```diff
@@ -2182,6 +2280,80 @@ hb_cluster_load_ping_host_list (char *ha_ping_host_list)
   return num_hosts;
 }
 
+/*
+ * hb_cluster_load_tcp_ping_host_list() -
+ *   return: number of tcp ping hosts
+ *
+ *   ha_ping_host_list(in):
+ */
+static int
+hb_cluster_load_tcp_ping_host_list (char *ha_ping_host_list)
+{
+  int num_hosts = 0;
+  char host_list[LINE_MAX];
+  char *host_list_p, *host_p, *host_pp;
+  char *port_p;
+  char buf[128];
+
+  if (ha_ping_host_list == NULL)
+    {
+      MASTER_ER_LOG_DEBUG (ARG_FILE_LINE, "ha_tcp_ping_hosts=NULL\n");
+      return 0;
+    }
+
+  strncpy_bufsize (host_list, ha_ping_host_list);
+
+  for (host_list_p = host_list;; host_list_p = NULL)
+    {
+      host_p = strtok_r (host_list_p, " ,", &host_pp);
+      if (host_p == NULL)
+	{
+	  break;
+	}
+
+      port_p = strstr (host_p, ":");
+      if (port_p == NULL || port_p == host_p)
+	{
+	  break;
+	}
+      else
+	{
+	  *port_p = '\0';
+
+	  do
+	    {
+	      port_p++;
+	    }
+	  while (*port_p == ' ');
+
+	  if (*port_p == '-')
+	    {
+	      break;
+	    }
+	}
+
+      if (strcmp (host_p, "0.0.0.0") == 0)
+	{
+	  snprintf (buf, sizeof (buf), "We do not allow 0.0.0.0 as a tcp ping hosts, excluded");
+	  MASTER_ER_SET (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_HB_NODE_EVENT, 1, buf);
+	}
+      else
+	{
+	  if (hb_add_tcp_ping_host (host_p, atoi (port_p)) == NULL)
```

Comment:
What happens if a user inputs a port as a string by accident like 1.1.1.1:3a3?

Reply:
1.1.1.1:3a3 becomes host_p="1.1.1.1", port_p="3a3", and the port number becomes 3 by the atoi() function. If you look at the attached test scenario in jira, this is the intended behavior. However, in the process of reviewing, I found points for improvement. It is going to be corrected.